internal/controllers/import_controller_test.go

```diff
@@ -54,6 +49,7 @@ var _ = Describe("reconcile CAPI Cluster", func() {
 		clusterRegistrationToken *managementv3.ClusterRegistrationToken
 		capiKubeconfigSecret     *corev1.Secret
 		clusterName              = "generated-rancher-cluster"
+		testObjectsSuffix        = "provisioningv1"
```

Comment:
Is this suffix required? It seems a bit out of place.

Reply:
Makes it easier to see which test this relates to i guess.

Reply:
With the existing configuration, running both `provisioning` and `management` tests would generate conflicts, as there are resources with the same name used by both test suites. This is used to keep them separated and avoid conflicts.

Reply:
I suspect these conflicts come from the fact we don’t create a temporary namespace for each test case. Envtest is very bad at removing things in time for the next test case. What you can use (also I’m working on introducing this in more places) is a new helper from `github.com/rancher-sandbox/rancher-turtles/internal/test/helpers`, which includes two things:
```go
		ns, err = testEnv.CreateNamespace(ctx, "ns-prefix") // create a unique ns
		Expect(err).ToNot(HaveOccurred())
// and to use in AfterSuite
                Expect(testEnv.Cleanup(ctx, ns, anyNumberOfObjects)).To(Succeed())
```

Reply:
I've tried to refactor the v3 tests with the new helpers and I'm still getting namespace collisions. This happens on the namespaces created from `import_sample.yaml`, specifically `cattle-system`, which I don't think we have control over during testing, other than modifying the yaml itself. Any ideas on whether this can be solved or we need to parse the yaml for each test?